### DISCUSSION TITLE
Simplify job entry

### DISCUSSION THREAD

**Discussion by johnlane** [O-X-L/ansible-webui-v0#8] [Ideas]
**It's very easy to enter a job**, however a few suggestions to make it even easier...

The only required information for a job is the playbook and inventory files, and the latter can also be defined in `ansible.cfg`. The job screen could be limited to those two fields only with an expander (similar to the "show logs" button?) to reveal the remaining fields.

If inventory is defined elsewhere, e.g. `ansible.cfg` then perhaps this could be displayed beneath the field which could then be allowed to be empty because inventory is defined elsewhere (of course, entering a value would override said default).

The fields list available files but you cannot select them (you either have to type them in or cut/paste). Maybe this is a quirk of my environment (browser maybe, but I tried two). It would be good if the drop-down was a selection rather than just a display.

Other defaults I think could be reversed: _schedule enabled_ and _needs credentials_ should default to _No_. They both require additional information to be supplied for their current default state so either that needs to be entered or those fields' values need to be changed. Reversing the defaults would remove the need to enter/change those fields in order to enter a job.

**Comment by ansibleguy** [comment]
Greetings.
Thanks for the feedback (:

1. Yeah - I will create an optimize form.
  https://github.com/ansibleguy/webui/issues/12

2. I was actually not aware that an inventory can be defined in the ansible.cfg - that will also be the case when using dynamic inventory plugins..
  Will change the behavior to not require an inventory for those cases!
  https://github.com/ansibleguy/webui/issues/11

3. Have thought of that - but did not yet find time to implement it. Will add it to the list (;
  https://github.com/ansibleguy/webui/issues/10

4. Reversing the credentials flag could make sense, yes. That one is a bit annoying if you do not need credentials. Is now in latest.
  The schedule-enabled flag is ignored if no schedule is supplied - so that one should be fine in my opinion. Maybe it could also be hidden if no schedule was supplied.

**Comment by ansibleguy** [comment]
If you have more ideas - I'm open to collaborate! (:
